src/jemalloc.c

```diff
@@ -2359,8 +2359,11 @@ je_malloc(size_t size) {
 			 * a large value.  If prof_active is set to true,
 			 * bytes_until_sample will be reset.
 			 */
```

Comment:
We need to update the comments. Any chance we can still keep the optimization (set to size_max) and find a way to reset it when prof active is changed?

Reply:
That means we'll need to go through all the threads and update their `bytes_until_sample` when `prof_active` is flipped from `false` to `true`?

Reply:
I’d prefer we don’t go that far. An alternative is to add a check on some slow path; anywhere you think would fit?

Reply:
How about we don't route to the slow path but simply continue on in the fast path (after increasing `bytes_until_sample` back)? After all there's no need to go to the slow path. The cost would be an extra `prof_active` check and an extra `bytes_until_sample` update.

Reply:
I feel adding a check on the slow path is doable, and a good choice can be the GC ticker path. (Other paths to slow path all seem to have the possibility of not being triggered after a very long time.) If we rely on the ticker, then the wrong `bytes_until_sample` can be fixed within a known number of malloc calls, though within these calls the total bytes allocated can still be huge.

Reply:
Yeah I think the complication of the slow path update isn’t worth it. 

We should still try optimizing a bit for the prof inactive case though, maybe set it to the sample interval? The current diff will frequently go down the reset path.

Reply:
Also are we double counting when going down malloc_default? It looks we already adjusted bytes until sample.

Reply:
I think the logic was added recent-ish in one of the fast path optimizations. @yinan1048576 let’s create an issue to track it (I bet it’s released
already).

Reply:
> We should still try optimizing a bit for the prof inactive case though, maybe set it to the sample interval? The current diff will frequently go down the reset path.

Great point! Doing that.

> Also are we double counting when going down malloc_default? It looks we already adjusted bytes until sample.

Yes we're double counting on the slow path. But I guess the impact would be minimal, since in the majority of the cases the execution won't go into the slow path. I guess there would also be some performance concerns, since we would probably need to do something extra on the fast path to indicate to the slow path that `bytes_until_sample` has already been set?

> let’s create an issue to track it (I bet it’s released already).

Yes it's released. Created #1605.